File: src/AnalyticsService/Extensions.fs
module AnalyticsService.Extensions

open AnalyticsService.Transport.Contracts.Requests
open AnalyticsService.Transport.Validation
open FluentValidation
open Microsoft.Extensions.DependencyInjection

type IServiceCollection with
    /// Extension method for registering request validators.
    member services.RegisterValidators () =
        services
            .AddScoped<IValidator<BatchStatRequest>, BatchStatRequestValidator>()

File: src/AnalyticsService/Extensions.cs
using AnalyticsService.Transport.Contracts.Requests;
using AnalyticsService.Transport.Validation;
using FluentValidation;

namespace AnalyticsService;

/// <summary>
/// A static class containing extension methods for <see cref="Program"/>.
/// </summary>
internal static class Extensions
{
    /// <summary>
    /// Extension method for registering request validators.
    /// </summary>
    public static IServiceCollection AddValidators(this IServiceCollection services)
    {
        return services
            .AddScoped<IValidator<BatchStatRequest>, BatchStatRequestValidator>()
            .AddScoped<IValidator<BatchPeriodStatRequest>, BatchPeriodStatRequestValidator>();
    }
}